cds/modules/deposit/api.py

```diff
@@ -787,6 +793,39 @@ def _tasks_global_status(self):
         iterate_events_results(events=events, fun=global_status)
         return global_status.status
 
+    def _rename_subtitles(self):
+        pattern = re.compile('(?P<filename>.*)_(?P<rest>[a-zA-Z]{2}\.vtt$)')
+        subtitles = CDSVideosFilesIterator.get_video_subtitles(self)
+        for subtitle_file in subtitles:
+            subtitle_obj = as_object_version(subtitle_file['version_id'])
+            match = pattern.match(subtitle_file['key'])
+            if match:
+                subtitle_obj.key = '{}_{}'.format(self['report_number'][0],
+                                                  match.group('rest'))
+                db.session.add(subtitle_obj)
+
+    def _publish_new(self, id_=None):
+        """Rename master file and subtitles before first publishing."""
+        id_ = id_ or uuid.uuid4()
+        self.mint_report_number(id_)
+        self['_files'] = self.files.dumps()
+
+        master_file = CDSVideosFilesIterator.get_master_video_file(self)
+        # This is needed because don't have always master file
+        if master_file:
+            master_obj = as_object_version(master_file['version_id'])
+            master_obj.key = '{}.{}'.format(self['report_number'][0], master_file.get('content_type') or
+                                            splitext(master_file['key'])[1][1:].lower())
+            db.session.add(master_obj)
+            self._rename_subtitles()
+        return super(CDSDeposit, self)._publish_new(id_)
+
+    def _publish_edited(self):
+        """Rename subtitles before publishing."""
```

Comment:
The same thing, could we emphasise that this function will publish, not just rename?

Reply:
This is method is not called when you replace the master file? Only the one above?

Reply:
Yep this method is called when editing a published video. But, if eventually subtitle names are  irrelevant with master file name in order to be working I will delete the subtitles renaming logic I think.